tests/performance/src/main/java/org/mule/runtime/core/processor/strategy/ProcessingStrategyBenchmark.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.processor.strategy;
+
+import static org.mule.runtime.core.api.lifecycle.LifecycleUtils.startIfNeeded;
+import static reactor.core.publisher.FluxSink.OverflowStrategy.ERROR;
+
+import org.mule.AbstractBenchmark;
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.construct.Flow;
+import org.mule.runtime.core.api.event.CoreEvent;
+import org.mule.runtime.core.api.processor.ReactiveProcessor;
+import org.mule.runtime.core.api.processor.Sink;
+import org.mule.runtime.core.api.processor.strategy.ProcessingStrategy;
+import org.mule.runtime.core.internal.processor.strategy.DirectProcessingStrategyFactory;
+import org.mule.runtime.core.internal.processor.strategy.TransactionAwareProactorStreamEmitterProcessingStrategyFactory;
+import org.mule.runtime.core.internal.processor.strategy.TransactionAwareProactorStreamWorkQueueProcessingStrategyFactory;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.infra.Blackhole;
+import org.reactivestreams.Publisher;
+
+import java.util.function.Function;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.FluxSink;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.MonoSink;
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.Throughput)
+public class ProcessingStrategyBenchmark extends AbstractBenchmark {
+
+  private MuleContext muleContext;
+
+  private ProcessingStrategy directPs;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove the "directPs" field and declare it as a local variable in the relevant methods. ([squid:S1450](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1450))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7657&issues=AWjJeTCLotIXnmsUTyuP&open=AWjJeTCLotIXnmsUTyuP)

[ISSUE_KEY]: AWjJeTCLotIXnmsUTyuP